### DISCUSSION TITLE
Buttons inside of an accessory

### DISCUSSION THREAD

**Discussion by iKrushYou** [HomeSpan/HomeSpan#349] [General]
To start, i have read a handful of discussions on custom characteristics so I know there are least some hurdles there

I was wondering if it is possible to add extra buttons inside of an accessory that could finely control the device.

I don't have it configured anymore, but for some reason I recall when i set up my Dyson fan on HomeBridge, it had a bunch of extra buttons inside for speed, oscillation, etc

For my example here, I have a blind (window covering) that I built, but sometimes it slips and gets out of sync. I would love to be able to have a button inside the accessory that would have it manually roll back up to reset the position

I know that anything we do here has to comply with the HAP standards, but I was hoping someone might have a sort of workaround here that could work. I have thought about creating a separate accessory to do this but I would end up with a ton of extra accessories that it wouldn't be very nice to look at / organize

**Comment by HomeSpan** [comment]
You can add one or more Switch Services to your Accessory that can be used to trigger other actions (like resyncing a position).  However, HomeKit does not include the concept of a simple push button, which is unfortunate since there are many uses for such a Service.  Instead, you'll need to emulate a push button by using a switch and having HomeSpan reset it to OFF a second or so after you've changed it to ON.  See the HomeSpan Tutorials section for a detailed example on exactly how to do this.

**Reply by iKrushYou** [reply]
Thanks so much for the response and the support. I will take a look at this later!
